vignettes/example.Rmd

```diff
@@ -0,0 +1,276 @@
+---
+title: "Example worklfow"
+author: 
+- Daniel Falster
+date: 2022
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{patch}
+  %\VignetteEncoding{UTF-8}
+  %\VignetteEngine{knitr::rmarkdown}
+editor_options: 
+  chunk_output_type: console
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+# installation
+
+```{r load libraries}
+library(plant)
+library(dplyr)
+library(tidyr)
+library(ggplot2)
+library(patchwork)
+```
+
+Activate logger
+
+```{r}
+plant_log_console()
+```
+
+Setting some base parameters.
+
+```{r}
+base_parameters <- function() {
+  p0 <- scm_base_parameters("FF16", "FF16_Env")
+
+  p0$strategy_default$lma <- 0.07
+  p0$strategy_default$hmat <- 15
+  p0$strategy_default$rho <- 700
+  p0$strategy_default$a_l1 <- 2.17
+  p0$strategy_default$a_l2 <- 0.5
+  
+  p0
+}
+```
+
+We can now create a function called `run_mypatch` which takes a number of a arguments, including `traits` which we assign a trait matrix. The `expand_parameters` function takes this trait matrix as an argument which allows us to simulate patches with multiple functional strategis, in addition to the base_parameters and any newly defined hyperparameters such as B_lf1. We also provide a latitude as an argument, as well as whether or not we want to run the patch to equilibrium seed rain.
+
+```{r}
+run_mypatch <- function(
+  traits =  trait_matrix(c(0.07), c("lma")),
+  B_lf1 = 1,
+  seed_rain = 100,
+  p0 = base_parameters(),
+  find_equilbrium = FALSE,
+  latitude = 28.182
+  ) {
+  
+  hyper_par_fn = make_FF16_hyperpar(B_lf1 = B_lf1, latitude = latitude)
+  p1 <- expand_parameters(traits, p0, hyper_par_fn, mutant = FALSE)
+  p1$seed_rain <- seed_rain
+  
+  if(find_equilbrium)
+    result <- equilibrium_seed_rain(p1)
+  else
+    result <- build_schedule(p1)
```

Comment:
This suggests we should always call `build_schedule` which might be a good idea, but means we don't get the blindingly fast results of the FF16 defaults. Would this work instead?

```r
run_my_patch <- function(
  ...
  find_equilibrium = FALSE,
  optimise_schedule = FALSE,
  ...
  
  if(find_equilbrium)
    result <- equilibrium_seed_rain(p1)
  else if(optimise_schedule)
    result <- build_schedule(p1)
  else
    result <- p1

  run_scm_collect(result)
}
```

Reply:
Good suggestion